src/mpl2/src/SimulatedAnnealingCore.cpp

```diff
@@ -272,10 +295,18 @@ void SimulatedAnnealingCore<T>::calWirelength()
   }
 
   for (const auto& net : nets_) {
-    const float x1 = macros_[net.terminals.first].getPinX();
-    const float y1 = macros_[net.terminals.first].getPinY();
-    const float x2 = macros_[net.terminals.second].getPinX();
-    const float y2 = macros_[net.terminals.second].getPinY();
+    T& source = macros_[net.terminals.first];
+    T& target = macros_[net.terminals.second];
+
+    if (target.isIOCluster()) {
```

Comment:
Will `source` never be a macro?

Reply:
Afaik, no. Perhaps add a critical for that case?

Reply:
I don\t think it is necessary. I was just curious why you're only checking if the `target` is an IO cluster, but if source is never an IO cluster, that's fine.
(My original comment was wrong, btw)